src/redux/features/missionSlice.js

```diff
@@ -0,0 +1,29 @@
+import { createSlice } from '@reduxjs/toolkit';
+import fetchMissionData from '../../apis/missionsApi';
+
+const initialState = {
+  missionData: [],
+  isLoading: false,
+};
+
+const missionSlice = createSlice({
+  name: 'mission',
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [fetchMissionData.pending]: (state) => {
+      state.isLoading = true;
+    },
+
+    [fetchMissionData.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      state.missionData = action.payload;
+    },
+
+    [fetchMissionData.rejected]: (state) => {
+      state.isLoading = false;
```

Comment:
Kindly, do not mutate the **`state`** object directly, instead return a new object consisting of the old state and the new changes.
For example, 
const currentState = {count: 0, age: 24};
const newState = {...currentState, age: 25}